tfhe/src/integer/gpu/ciphertext/mod.rs

```diff
@@ -389,6 +389,66 @@ impl CudaRadixCiphertextInfo {
                 .collect(),
         }
     }
+
+    pub(crate) fn after_extend_radix_with_trivial_zero_blocks_lsb(
+        &self,
+        num_blocks: usize,
+    ) -> Self {
+        let mut new_block_info = Self { blocks: Vec::new() };
+        for _ in 0..num_blocks {
+            new_block_info.blocks.push(CudaBlockInfo {
+                degree: Degree::new(0),
+                message_modulus: self.blocks.first().unwrap().message_modulus,
+                carry_modulus: self.blocks.first().unwrap().carry_modulus,
+                pbs_order: self.blocks.first().unwrap().pbs_order,
+                noise_level: NoiseLevel::ZERO,
+            });
+        }
+        for &b in self.blocks.iter() {
+            new_block_info.blocks.push(b);
+        }
+        new_block_info
+    }
+
+    pub(crate) fn after_extend_radix_with_trivial_zero_blocks_msb(
+        &self,
+        num_blocks: usize,
+    ) -> Self {
+        let mut new_block_info = Self { blocks: Vec::new() };
+        for &b in self.blocks.iter() {
+            new_block_info.blocks.push(b);
+        }
+        for _ in 0..num_blocks {
+            new_block_info.blocks.push(CudaBlockInfo {
+                degree: Degree::new(0),
+                message_modulus: self.blocks.first().unwrap().message_modulus,
+                carry_modulus: self.blocks.first().unwrap().carry_modulus,
+                pbs_order: self.blocks.first().unwrap().pbs_order,
+                noise_level: NoiseLevel::ZERO,
+            });
+        }
+        new_block_info
+    }
+
+    pub(crate) fn after_trim_radix_blocks_lsb(&self, num_blocks: usize) -> Self {
+        let mut new_block_info = Self { blocks: Vec::new() };
+        for (i, &b) in self.blocks.iter().enumerate() {
+            if i >= num_blocks {
+                new_block_info.blocks.push(b);
+            }
+        }
+        new_block_info
+    }
+
+    pub(crate) fn after_trim_radix_blocks_msb(&self, num_blocks: usize) -> Self {
+        let mut new_block_info = Self { blocks: Vec::new() };
+        for (i, &b) in self.blocks.iter().enumerate() {
+            if i < num_blocks {
+                new_block_info.blocks.push(b);
+            }
+        }
+        new_block_info
```

Comment:
same idea as above, we can use Vec::with_capacity, and avoid the if by slicing `self.blocks`